mvrx/src/main/kotlin/com/airbnb/mvrx/CoroutinesStateStore.kt

```diff
@@ -0,0 +1,114 @@
+package com.airbnb.mvrx
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import java.util.concurrent.Executors
+
+@Suppress("EXPERIMENTAL_API_USAGE")
+class CoroutinesStateStore<S : MvRxState>(initialState: S) : MvRxStateStore<S> {
+
+    /** Channel that serves as a trigger to flush the setState and withState queues. */
+    private val flushQueuesChannel = Channel<Unit>(capacity = Channel.CONFLATED)
+    private val setStateChannel = Channel<S.() -> S>(capacity = Channel.UNLIMITED)
+    private val withStateChannel = Channel<(S) -> Unit>(capacity = Channel.UNLIMITED)
+
+    private val stateChannel = ConflatedBroadcastChannel<S>(initialState)
```

Comment:
ConflatedBroadcastChannel can drop events. Is it expected behavior for CoroutinesStateStore?
Real example of dropped events can be state updates for `Async`. 

```kotlin
    @Test
    fun test_ConflatedBroadcastChannel_runBlocking() {
        runBlocking {
            val store = CoroutinesStateStore(MvRxStateStoreTestState())
            async {
                store.flow.collect {
                    println("received count=${it.count}")
                }
            }
            async {
                (1..5).forEach {
                    println("send count=$it")
                    store.set { copy(count = it) }
                }
                store.cancel()
            }
        }
        // console out
        // received count=1
        // send count=1
        // send count=2
        // send count=3
        // send count=4
        // send count=5
        // received count=5
    }
```

Reply:
@rougsig great catch. It is fixed with a buffer in the latest code.